sphinx_gallery/gen_rst.py

```diff
@@ -351,6 +363,7 @@ def save_figures(image_path, fig_count, gallery_conf):
         current_fig = image_path.format(fig_count + fig_mngr.num)
         fig.savefig(current_fig, **kwargs)
         figure_list.append(current_fig)
+        fig._sg_captured = True
```

Comment:
Yeah I could do that. I don't think indexing will work:
```
>> import matplotlib.pyplot as plt
>>> fig = plt.figure()
>>> fig.number
1
>>> plt.close('all')
>>> fig = plt.figure()
>>> fig.number
1
```
Instead I could maintain a list and check with `is`. The problem with that approach is that it might lead to memory issues, because the figures can't be garbage-collected even if they are closed by the user. So I'm actually starting to think that the monkey patching, although ugly, might be better.